### PR TITLE
Update FindBugs annotation library and maven plugin

### PR DESCRIPTION
**Pull request by KostyaSha**
I tried follow version 1.0.0 version and found that it replaced with other library. But it fails because of bytecode. Investigating it, will keep it open here.

### CODE REVIEW

**Comment by KostyaSha** [comment]
Will back when jenkins migrate on java7 :)

**Comment by KostyaSha** [comment]
Now with java7 it much easier to do update :)

**Comment by KostyaSha** [comment]
re-triggerring again, because https://jenkins.ci.cloudbees.com/job/core/job/jenkins-core/2700/testReport/junit/jenkins.security/FilePathSecureTest/tar/ failure unrelated i think.

File: core/pom.xml [hunk 1]
```diff
@@ -869,7 +869,7 @@ THE SOFTWARE.
           <plugin>
             <groupId>org.codehaus.mojo</groupId>
             <artifactId>findbugs-maven-plugin</artifactId>
-            <version>2.5.2</version>
+            <version>3.0.0</version>
```
Comment (jglick) [inline core/pom.xml]: But you could override that in this PR in (root) `pom.xml`.

**Comment by jglick** [comment]
This got clobbered by your update:

> But you could override [`findbugs-maven-plugin.version`] in this PR in (root) `pom.xml`.

**Comment by jglick** [comment]
Other than the duplication, :+1:

**Comment by KostyaSha** [comment]
@jglick maybe remove reporting at all? Or somebody using it?

**Comment by KostyaSha** [comment]
maven-site is not updated https://github.com/jenkinsci/maven-site

**Comment by KostyaSha** [comment]
@jglick set with variable, but can't test because 
```
[ERROR] Failed to execute goal on project jenkins-core: Could not resolve dependencies for project org.jenkins-ci.main:jenkins-core:jar:1.618-SNAPSHOT: Could not find artifact org.jenkins-ci.main:cli:jar:1.618-SNAPSHOT in repo.jenkins-ci.org (http://repo.jenkins-ci.org/public/) -> [Help 1]
```
I think nobody uses mvn site and better add build execution for metrics/findbugs profiles.

**Comment by jglick** [comment]
I do not think `mvn site` is used.

File: pom.xml [hunk 2]
```diff
@@ -501,7 +501,7 @@ THE SOFTWARE.
         <plugin>
           <groupId>org.apache.maven.plugins</groupId>
           <artifactId>maven-site-plugin</artifactId>
-          <version>3.1</version>
+          <version>3.3</version>
```
Comment (jglick) [inline pom.xml]: Could probably be reverted.

**Comment by jglick** [comment]
:+1:; is this still supposed to WiP?

File: pom.xml (hunk 2 above)
Comment (KostyaSha) [inline pom.xml]: No, either mvn site will fail with absent classes.

Comment (KostyaSha) [inline pom.xml]:
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-site-plugin:3.1:site (default-site) on project pom: Execution default-site of goal org.apache.maven.plugins:maven-site-plugin:3.1:site failed: A required class was missing while executing org.apache.maven.plugins:maven-site-plugin:3.1:site: org/sonatype/aether/graph/DependencyFilter
[ERROR] -----------------------------------------------------
```

**Comment by KostyaSha** [comment]
@jglick well i not sure that anybody using this reporting at all. I can suggest move findbugs execution from reporting to some build goal.

**Comment by KostyaSha** [comment]
As update for current versions this PR done.

**Comment by oleg-nenashev** [comment]
:+1:

**Pull request merged**